### PR TITLE
Remove `textScaleFactor` dependent logic from `AppBar`

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
I am trying to remove `textScaleFactor`-dependent logic from the framework since it's likely going to be deprecated, hopefully the original logic isn't from the material spec.

I stole the sample code from https://github.com/flutter/flutter/pull/125038 and here are the screenshots (`textScaleFactor = 3.0`). 

Internal Tests: **no relevant test failures**

|  Medium  |   Large |
| --------------- | --------------- |
| flutter_01 | flutter_03 |

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by guidezpl** [comment]
For context, a lot of text scale fixes were introduced ~3 years ago (b/hotlists/2302728) to address overscaling of text in certain components. It's not been reintegrated in spec (I believe) due to [this](https://docs.google.com/document/d/1MT9v9CX4q42lV-Piu6aeUXBKbIXy2bLJRAlFiF6BWN0) coming up, which is likely to negate the need for special handling for that platform, at least.

File: packages/flutter/lib/src/material/app_bar.dart [hunk 1]
```diff
@@ -2081,79 +2079,48 @@ class _RenderAppBarTitleBox extends RenderAligningShiftedBox {
...
-        return EdgeInsets.fromLTRB(
-          padding.left,
-          0,
-          padding.right,
-          adjustBottomPadding / textScaleFactor,
-        );
-      }
-      return padding;
-    }
+    final Widget? expandedTitle = switch ((title, expandedTextStyle)) {
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar.dart:2114]: This is slick but I'm not sure the logic is the same or even simpler than before.
```dart
    if (title != null) {
      ...
      expandedTitle = config.expandedTextStyle != null
        ? DefaultTextStyle(
            style: expandedTextStyle!,
            child: title!,
          )
        : title;
    }
 ```

Would only wrap the title with the DefaultTextStyle if `config.expandedTextStyle` was non-null. The new code checks `expandedTextStyle` which incorporates the widget/theme defaults.  I don't know why the code was the way it was, so I suppose if the change doesn't break anything it's OK.

**Review by HansMuller — APPROVED** [review]
This LGTM although I'm only assuming that the layout remains compatible because the existing tests pass.

File: packages/flutter/lib/src/material/app_bar.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/app_bar.dart:2114]: It's from https://github.com/flutter/flutter/pull/122542/files#diff-20c19d254b0a82027625f1246ba51cb2264bfabf105fe91d532c19e6a6d77478L2141-L2151

@TahaTesser does this change look ok to you?

Comment (TahaTesser) [reply packages/flutter/lib/src/material/app_bar.dart:2114]: This looks good to me! 

`config.expandedTextStyle != null` existed before I added `title ?? appBarTheme.titleTextStyle`. I should have updated it to just `expandedTextStyle != null`.  Yours refactor makes it much cleaner and hard to miss.

File: dev/tools/gen_defaults/lib/app_bar_template.dart [hunk 2]
```diff
@@ -79,7 +79,7 @@ class _MediumScrollUnderFlexibleConfig with _ScrollUnderFlexibleConfig {
     ${textStyle('md.comp.top-app-bar.medium.headline')}?.apply(color: ${color('md.comp.top-app-bar.medium.headline.color')});
 
   @override
-  EdgeInsetsGeometry? get expandedTitlePadding => const EdgeInsets.fromLTRB(16, 0, 16, 20);
+  EdgeInsetsGeometry get expandedTitlePadding => const EdgeInsets.fromLTRB(16, 0, 16, 20);
```
[the remark this answers is not in the document]
Comment (TahaTesser) [reply dev/tools/gen_defaults/lib/app_bar_template.dart:82]: +1

File: packages/flutter/lib/src/material/app_bar.dart [hunk 3]
```diff
@@ -2081,79 +2079,48 @@ class _RenderAppBarTitleBox extends RenderAligningShiftedBox {
...
+  final _FlexibleConfigBuilder configBuilder;
   final TextStyle? titleTextStyle;
   final double bottomHeight;
 
   @override
   Widget build(BuildContext context) {
-    late final ThemeData theme = Theme.of(context);
     late final AppBarTheme appBarTheme = AppBarTheme.of(context);
-    final AppBarTheme defaults = theme.useMaterial3 ? _AppBarDefaultsM3(context) : _AppBarDefaultsM2(context);
+    late final AppBarTheme defaults = Theme.of(context).useMaterial3 ? _AppBarDefaultsM3(context) : _AppBarDefaultsM2(context);
```
Comment (TahaTesser) [inline packages/flutter/lib/src/material/app_bar.dart:2100]: This is inside `build`. We might not need `late`.

```suggestion
   final AppBarTheme defaults = Theme.of(context).useMaterial3 ? _AppBarDefaultsM3(context) : _AppBarDefaultsM2(context);
```

**Review by TahaTesser — APPROVED** [review]
Thanks for doing this. @LongCatIsLooong 

LGTM with minor nits.

File: packages/flutter/lib/src/material/app_bar.dart (hunk 3 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/app_bar.dart:2100]: The `late` keywords are for lazy evaluation. When `foregroundColor` is specified, `defaults` won't be evaluated so this widget doesn't have to depend on the `Theme` widget.

**Review by guidezpl — APPROVED** [review]

**Pull request merged**